src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T extends object>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const groupedMap: GroupsMap<T> = {};
+
+  items.map((item) => {
+    const keyValue: any = item[key];
+    const keyString = String(keyValue);
+
+    if (Object.prototype.hasOwnProperty.call(groupedMap, keyString)) {
```

Comment:
Why not `if (groupedMap.hasOwnProperty(keyString))` or simply `if (groupedMap[keyString])`?

Reply:
I was studying some new things, but your approach is way simple, thanks!